### PR TITLE
Mod metadata upgrade

### LINKED ISSUE #6: Give patched mods missing an icon a greyed-out Patchwork icon
This would make more sense as opposed to a greyed-out fabric icon.

We'll need an icon from @Prospector for this!

### PR DESCRIPTION
**Pull request by TheGlitch76**
Mods now use greyed Patchwork icon when one is missing, mark themselves as Forge mods in ModMenu, and multi-mod JARs now work again. Closes #6

### CODE REVIEW

File: src/main/java/com/patchworkmc/Patchwork.java [hunk 1]
```diff
@@ -45,15 +43,23 @@
...
 
 public class Patchwork {
 	public static final Logger LOGGER;
 	private static String version = "1.14.4";
 
+	private static byte[] patchworkIcon;
+
 	static {
+		try {
+			patchworkIcon = Files.readAllBytes(new File(Patchwork.class.getResource("/patchwork-greyscale.png").getPath()).toPath());
```
Comment (coderbot16) [inline src/main/java/com/patchworkmc/Patchwork.java]: maybe call it `patchwork-icon-greyscale.png`?

File: src/main/java/com/patchworkmc/Patchwork.java [hunk 2]
```diff
@@ -213,44 +221,66 @@ public static void transformMod(Path currentPath, Path jarPath, Path outputRoot,
...
-			ByteArrayOutputStream jar = new ByteArrayOutputStream();
-			ZipOutputStream zip = new ZipOutputStream(jar);
+			// generate the jar
+			Path subJarPath = Paths.get("temp/" + modid + ".jar");
+			Map<String, String> env = new HashMap<>();
+			env.put("create", "true");
+			FileSystem subFs = FileSystems.newFileSystem(new URI("jar:" + subJarPath.toUri().toString()), env);
+
+			// Write patchwork logo
+			if (entry.getAsJsonPrimitive("icon").getAsString().equals("assets/patchwork-generated/icon.png")) {
```
Comment (coderbot16) [inline src/main/java/com/patchworkmc/Patchwork.java]: Deduplicate this code please

File: src/main/java/com/patchworkmc/Patchwork.java [hunk 3]
```diff
@@ -213,44 +221,66 @@ public static void transformMod(Path currentPath, Path jarPath, Path outputRoot,
...
+		}
+
 		try {
 			Files.createDirectory(fs.getPath("/META-INF/jars/"));
 		} catch (IOException ignored) {
 			// ignored
 		}
 
+		// If you touch this, you better be SURE it works properly before pushing.
+		// Yes, it's a hack, but it works.
```
Comment (coderbot16) [inline src/main/java/com/patchworkmc/Patchwork.java]: This doesn't really qualify as a hack anymore, it's just normal zipfs usage tbh

File: src/main/java/com/patchworkmc/Patchwork.java [hunk 4]
```diff
@@ -213,44 +221,66 @@ public static void transformMod(Path currentPath, Path jarPath, Path outputRoot,
...
 				continue;
 			}
 
-			ByteArrayOutputStream jar = new ByteArrayOutputStream();
-			ZipOutputStream zip = new ZipOutputStream(jar);
+			// generate the jar
+			Path subJarPath = Paths.get("temp/" + modid + ".jar");
+			Map<String, String> env = new HashMap<>();
+			env.put("create", "true");
+			FileSystem subFs = FileSystems.newFileSystem(new URI("jar:" + subJarPath.toUri().toString()), env);
```
Comment (coderbot16) [inline src/main/java/com/patchworkmc/Patchwork.java:245]: Would it be possible to create a nested zipfs?

**Review by coderbot16 — CHANGES_REQUESTED** [review]
Overall looks good, a few things that need to get looked at though.

File: src/main/java/com/patchworkmc/Patchwork.java (hunk 4 above)
Comment (TheGlitch76) [reply src/main/java/com/patchworkmc/Patchwork.java:245]: I tried that in my original implementation a long time ago and it didn't work

**Review by coderbot16 — APPROVED** [review]
Thanks for this!

**Pull request merged**